### PR TITLE
Simplify runtime api error handling

### PR DESCRIPTION
**Pull request by bkchr**
This pr simplifies the runtime api error handling by introducing an error type for the runtime api.

polkadot companion: https://github.com/paritytech/polkadot/pull/2437

### CODE REVIEW

**Review by ordian — APPROVED** [review]
I don't fully understand the `UnwindSafe` bounds and UI tests removal reasoning, but the change from `String` to proper error type looks good.

File: client/block-builder/src/lib.rs [hunk 1]
```diff
@@ -237,7 +232,7 @@ where
 				ExecutionContext::BlockConstruction,
 				inherent_data
 			))
-		})
+		}).map_err(|e| Error::Application(Box::new(e)))
```
Comment (drahnr) [inline client/block-builder/src/lib.rs:235]: I would prefer adding another error variant, so `::Application` can cease to exist eventually.

**Review by drahnr — APPROVED** [review]
This looks very good!

**Comment by bkchr** [comment]
bot merge

**Pull request merged**